### ISSUE TITLE
Need a LICENSE file

### ISSUE THREAD

**Issue by Moelf** [madjax-hep/madjax#17]
hesitant to read the source code now cuz I don't understand what https://github.com/restrepo/madgraph/blob/master/madgraph/LICENSE is and if it's compatible both ways with licenses like MIT which would wish my package to be in.

**Comment by matthewfeickert** [comment]
@Moelf `madjax` does indeed need a LICENSE, and given

https://github.com/madjax-hep/madjax/blob/202842712df69aa8d961c3f894991e6ae1f8b96e/setup.py#L53

it seems that it should be Apache.

Though `madjax`'s license is not contingent on MadGraph5_aMC@NLO's 

```console
$ docker run --rm scailfin/madgraph5-amc-nlo:mg5_amc3.3.1 'cat /usr/local/venv/MG5_aMC/LICENSE'
The MadGraph5_aMC@NLO software comes with the following Open Source License, 
adapted from the University of Illinois/NCSA license.

Copyright (c) 2009, 2013, the MadTeam.  All rights reserved.

Developed by the "MadTeam", i.e. 

Johan ALWALL
Rikkert FREDERIX
Stefano FRIXIONE
Michel HERQUET
Valentin HIRSCHI
Fabio MALTONI
Olivier MATTELAER
Tim STELZER
Paolo TORRIELLI
Marco ZARO

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to
deal with the Software without restriction, including without limitation the
rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
sell copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:
  1. Redistributions of source code must retain the above copyright notice,
     this list of conditions and the following disclaimers.
  2. Redistributions in binary form must reproduce the above copyright
     notice, this list of conditions and the following disclaimers in the
     documentation and/or other materials provided with the distribution.
  3. Neither the name "MadTeam", the names of the authors, their institution, 
     nor the names of its contributors may be used to endorse or promote 
     products derived from this Software without specific prior written 
     permission.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
WITH THE SOFTWARE.
```

All of `madjax`'s code that interacts with MadGraph5 is still `madjax` and the MadGraph5_aMC@NLO license doesn't place any restrictions on hooking into it given

> to deal with the Software without restriction

**Comment by Moelf** [comment]
I assumed in order to make ME calculation, someone read madgraph source code at some point. Maybe that's not how this package works. Either way thanks for the prompt response

**Comment by matthewfeickert** [comment]
> I assumed in order to make ME calculation differentiable, someone had to read madgraph source code at some point to re-implement some parts of it in jax. Maybe that's not how this package works.

I'll let Lukas or Michael who actually did the work comment there, but my understanding from (the abstract of) https://arxiv.org/abs/2203.00057 is that plugins from [`src/madjax/mg5/madjax_me_gen`](https://github.com/madjax-hep/madjax/tree/202842712df69aa8d961c3f894991e6ae1f8b96e/src/madjax/mg5/madjax_me_gen) get injected into MadGraph5's plugin system to steer the code generation that in addition to the Fortran and other stuff also writes hooks for JAX, so you can later access gradient information. So I think(?) it is more "injecting code into MadGraph5 to get it to generate code for you" than "re-implement[ing] ... parts of it". Safe to say that I don't know much here. :slightly_smiling_face: 

But I think it is pretty safe to say that `madjax` isn't beholden to `MadGraph5_aMC@NLO`'s LICENSE. (I don't have a good comparison example though from other areas of software, as `UnROOT.jl` and `uproot` both are just reading/writing a binary file format, which is a bit different from using a plugin system. Anyone reading this please insert a better analogy here. :slightly_smiling_face:)

**Comment by Moelf** [comment]
ROOT's license is such that it's compatible in both ways with uproot's (BSD-3) and UnROOT's (MIT).

There's actually another step in that story, uproot's probably also has constrained from go-hep/hep since some of the reverse engineerings was learned from go, but maybe that didn't involve reading source code of go-hep/.

but yeah, if `madjax` is just using MadGraph's plugin system, this is totally free to choose whatever LICENSE.
